test/sample_consensus/test_sample_consensus_quadric_models.cpp

```diff
@@ -824,6 +826,97 @@ TEST (SampleConsensusModelCylinder, projectPoints)
   EXPECT_XYZ_NEAR(projected_points_all[7],        (*input)[7], 1e-5);
 }
 
+TEST(SampleConsensusModelEllipse3D, RANSAC)
+{
+  srand(0);
+
+  // Using a custom point cloud on a tilted plane
+  PointCloud<PointXYZ> cloud;
+  cloud.resize(22);
+
+  cloud[ 0].getVector3fMap() << 0.9996, 5.0000, 2.9996;
+  cloud[ 1].getVector3fMap() << 0.6916, 5.0000, 2.9022;
+  cloud[ 2].getVector3fMap() << 0.4124, 5.0000, 2.6181;
+  cloud[ 3].getVector3fMap() << 0.1909, 5.0000, 2.1757;
+  cloud[ 4].getVector3fMap() << 0.0485, 5.0000, 1.6177;
+  cloud[ 5].getVector3fMap() << -0.0007, 5.0000, 0.9998;
+  cloud[ 6].getVector3fMap() << 0.0485, 5.0000, 0.3818;
+  cloud[ 7].getVector3fMap() << 0.1903, 5.0000, -0.1746;
+  cloud[ 8].getVector3fMap() << 0.4118, 5.0000, -0.6189;
+  cloud[ 9].getVector3fMap() << 0.6902, 5.0000, -0.9023;
+  cloud[10].getVector3fMap() << 1.0007, 5.0000, -0.9994;
+  cloud[11].getVector3fMap() << 1.3100, 5.0000, -0.9019;
+  cloud[12].getVector3fMap() << 1.5872, 5.0000, -0.6184;
+  cloud[13].getVector3fMap() << 1.8094, 5.0000, -0.1748;
+  cloud[14].getVector3fMap() << 1.9501, 5.0000, 0.3812;
+  cloud[15].getVector3fMap() << 2.0002, 5.0000, 0.9996;
+  cloud[16].getVector3fMap() << 1.9514, 5.0000, 1.6188;
+  cloud[17].getVector3fMap() << 1.8081, 5.0000, 2.1759;
+  cloud[18].getVector3fMap() << 1.5868, 5.0000, 2.6184;
+  cloud[19].getVector3fMap() << 1.3083, 5.0000, 2.9021;
```

Comment:
It turns out that this old data I used had noise added to it (on purpose, for the code I had written at that time). Now with noise-free data, the test runs smoothly.